### PR TITLE
[BE]: Follow detach().clone() pattern for SGD

### PR DESCRIPTION
**Pull request by Skylion007**
Clone() copies the gradients too, but we immediately detach them. Detach returns a view of the tensor without it's gradients, and the copies only that subset. Related to #144270

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
Okay, though practically speaking this is likely not changing much perf in most use cases as grad normally does not require grad.

**Comment by Skylion007** [comment]
@janeyx99 Seems like a false positive, but I would like to get confirmation

**Comment by janeyx99** [comment]
Ah I think you do have to modify the test now that the version counts are different.

File: torch/optim/sgd.py [hunk 1]
```diff
@@ -352,7 +352,7 @@ def _single_tensor_sgd(
             buf = momentum_buffer_list[i]
 
             if buf is None:
-                buf = torch.clone(grad).detach()
+                buf = torch.clone(grad.detach())
```
Comment (cyyever) [inline torch/optim/sgd.py:355]: Use `grad.detach().clone` ? Since that is a common pattern in the code base.

File: torch/optim/sgd.py [hunk 2]
```diff
@@ -441,7 +441,7 @@ def _multi_tensor_sgd(
                     if device_momentum_buffer_list[i] is None:
                         buf = device_momentum_buffer_list[i] = momentum_buffer_list[
                             indices[i]
-                        ] = torch.clone(device_grads[i]).detach()
+                        ] = torch.clone(device_grads[i].detach())
```
Comment (cyyever) [inline torch/optim/sgd.py:444]: Same as the above.

**Review by cyyever — APPROVED** [review]
Are there still undiscovered clones like these?

**Comment by albanD** [comment]
> Clone() copies the gradients too

That is NOT true.
torch.clone() and Tensor.clone() do NOT clone the gradients.
Maybe you're thinking of deepcopy(), which indeed does?

**Review by janeyx99 — COMMENTED** [review]
As clone does not copy gradients, can you update the PR description? We can still follow @cyyever's suggestions here to land cleaner code for BE, even though it will only be marginally better.
